### ISSUE TITLE
Status of 22.03 release

### ISSUE THREAD

**Issue by lukasz1992** [ericpaulbishop/gargoyle#972]
Hi,

that is not really a bug nor issue, I just like to know what are the plans for new release or answers to these questions:
- is there any roadmap or a list of features/changes/goals that new release is going to have
- what are the plans for: switching to ksmbd, switching to kernel ntfs3 driver (drop fuse and ntfs-3g), wireguard and support for Wifi 6/6E?
- what tests need to be done before the release

**Comment by obsy** [comment]
See branch: https://github.com/ericpaulbishop/gargoyle/commits/base_on_openwrt_2203

**Comment by lukasz1992** [comment]
@obsy this answers to my second question only:
- ksmbd and wireguard are already present; ntfs3-3g is still used in 22.03 (but could be backported), and WiFi 6E is not supported yet

**Comment by lantis1008** [comment]
Ive already switched to ksmbd by default (samba36 is still available for those that want to compile it). 
I won't backport ntfs3. Sorry it's not worth the effort. We will pick it up in OpenWrt 23.xx I expect. 

Features:
- Based on OpenWrt 22.03
- New diagnostics plug-in
- New targets (ipq40xx, mediatek, ramips.mt76x8, and potentially 2 more)
- As many devices supported as possible
- WiFi 6 (ax) support on 5GHz only
- Quotas can now be arbitrarily set/reset via the GUI
- A handful of bug fixes

Roadmap/plans:
- Get Gargoyle building on the latest OpenWrt
- Fix up Qosmon bugs
- Support WiFi 6 on 2.4GHz
- Bring in bcm27xx and rockchip targets (pending PR and feedback on forum)

Testing:
- everything. User feedback is desperately needed, and a forum post has been made https://www.gargoyle-router.com/phpbb/viewtopic.php?t=17779

**Comment by obsy** [comment]
proposal: drop old atheros and ramips/rt305x. These are old devices, slow and often do not have enough ram (or flash)

**Comment by lantis1008** [comment]
Agreed, but it has been so long since the last release that i'd like to keep them for this one. Give them a final send off.
They will be dropped for 23.xx

**Comment by lukasz1992** [comment]
Another proposal would be to extend available modes for WiFi.

Currently we can select b+g+n or b+g. I do not think anyone would need b+g in default config.

So my proposal - is to allow such options:
b+g+n (or g+n), g+n+ax, n+ax

And for 5GHz:
an+ac, an+ac+ax (or an+ac on WiFi 5 devices) or ac+ax (or ac only on WiFi 5 devices)

**Comment by obsy** [comment]
These options are deprecated (see hwmode in  https://openwrt.org/docs/guide-user/network/wifi/basic )

We can set require_mode (n or ac)

**Comment by lukasz1992** [comment]
and other settings (not checked if already done):
- enabling short_preamble
- setting legacy_rates=0 for disabling 802.11b
- basic_rate for require_mode=n or ac should be also tuned
- enabling ieee80211w to 1 (or 2 for WPA3/OWE)
- sae_pwe to 2 for WPA3

**Comment by lantis1008** [comment]
I don't see large benefit in tweaking a lot of these parameters for most users. 
legacy rates is already turned off, so strictly speaking the GUI is misleading. I don't see that being critical. 
N, AC or AX only modes? Pretty niche. 
Maybe for the next release. The entire connection basic page needs a rework anyway. It's become a terror to work on.

**Comment by pricerc** [comment]
> I don't see large benefit in tweaking a lot of these parameters for most users. legacy rates is already turned off, so strictly speaking the GUI is misleading. I don't see that being critical. N, AC or AX only modes? Pretty niche. Maybe for the next release. The entire connection basic page needs a rework anyway. It's become a terror to work on.

I'd imagine most users just want to set their SSID and password and see the rest as noise.

**Comment by lantis1008** [comment]
Exactly. 
It might be something worth exposing on the new Connection->Advanced page, however.

**Comment by pricerc** [comment]
I do miss my Gargoyle router (there's an ongoing discussion over on the Gargoyle forums about availability in NZ) - I gave mine up when my teenagers had left home and I found myself needing ipv6.

But I do have an OpenWRT router with Luci, which is a good example of how *not* to edit your router's wifi settings:
- the SSID and password are on separate tabs that you have to go looking for. 
- at the top of the 'edit' screen, there is a 'status' window; if you're going to have one on an 'edit' screen, it should be at the bottom.
- that is followed by a channel selector, which most people wouldn't know what to do with, because to understand the implications of changing them would require research...
- and they have country selection on the 'advanced' tab, when I would have thought it's the first thing you should be making sure is correct.

Clearly designed by people who see user experience as way down on their list of priorities.

**Comment by lukasz1992** [comment]
I am for tuning WiFi settings in Connection -> Advanced page:
- move txpower settings there
- move settings to enable WPA2-enterprise (I think it should be hidden in basic options) - this is an advanced config
- allow to select required WiFi version per band: WiFi 3(default), WiFi 4 and WiFi 5 for 5GHz (also allow to force Greenfield)
- allow to increase 802.11w setting (like set it to required for WPA2 only mode)
- allow to set cell_density
I could implement those in this month, but need help with translations for other languages

**Comment by lantis1008** [comment]
> btw why WPA3-Enterprise settings are commented out - no testers?

Correct

**Comment by obsy** [comment]
@lukasz1992: Can be closed - version 1.14.0 based on Openwrt 22.03 has been released.
